Sources/SwiftLexicalLookup/LookupState.swift

```diff
@@ -0,0 +1,21 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+@_spi(Experimental) public struct LookupState {
```

Comment:
I added an appropriate doc comment. 

On a side note, thanks to @ahoppen suggestion, I've simplified how we handle lookup from inside `guard ... else` body to [filter those inside `introducesToSequentialParent` in `guard` scope implementation](https://github.com/swiftlang/swift-syntax/blob/252d3be80c010add2bac9f9712ac32d9892ea574/Sources/SwiftLexicalLookup/ScopeImplementations.swift#L321-L322). 

Do you think we should remove `LookupState` now, as it doesn't provide any meaningful functionality, or should we leave it as is. It could potentially be beneficial in the future to have it already accessible across the lookup API in case we want to utilize it.

Reply:
If we don't need it now, let's remove it! The fewer parameters and less state we need to track, the better. If we find we need to bring it back... well, we know how to do it.

Reply:
I agree that we should remove `LookupState` altogether.

Reply:
Ok. I removed `LookupState` and the internal `_lookup` method.